lib/spack/spack/build_systems/autotools.py

```diff
@@ -151,9 +151,32 @@ def configure(self, spec, prefix):
         options = ['--prefix={0}'.format(prefix)] + self.configure_args()
         inspect.getmodule(self).configure(*options)
 
+    def make_targets_test(self):
+        """Should return an iterable containing all the targets that
+        must be passed to `make` to check the build.
+
+        Default: look for 'check' and 'test'
+        """
+        targets = []
+
+        for default_test_target in ['test', 'check']:
+            if _make_target_exists(default_test_target):
+                targets.append(default_test_target)
+
+        return targets
+
+    def make_targets_build(self):
```

Comment:
I am instead in favor of coding an hook to generalize the targets for `build`. Anyhow I would change the name to `build_targets`. For symmetry reason I would add a similar `install_targets` method.